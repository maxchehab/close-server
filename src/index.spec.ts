import http from 'http';
import closeServer from '.';

describe('close-server', () => {
  it('throws error when closing server that did not run', async () => {
    const srv = http.createServer();
    await expect(closeServer(srv)).rejects.toMatchSnapshot();
  });

  it('throws error when closing server that already is closed', async () => {
    const srv = http.createServer();
    srv.listen();

    await closeServer(srv);
    await expect(closeServer(srv)).rejects.toMatchSnapshot();
  });

  it('closes a running server successfully', async () => {
    const srv = http.createServer();
    srv.listen();

    expect(await closeServer(srv)).toBeUndefined();
    expect(srv.listening).toEqual(false);
  });
});
