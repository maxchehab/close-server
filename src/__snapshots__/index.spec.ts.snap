// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`close-server throws error when closing server that already is closed 1`] = `[Error [ERR_SERVER_NOT_RUNNING]: Server is not running.]`;

exports[`close-server throws error when closing server that did not run 1`] = `[Error [ERR_SERVER_NOT_RUNNING]: Server is not running.]`;
